cmd/krew/cmd/install.go

```diff
@@ -62,6 +99,20 @@ Remarks:
 			var pluginNames = make([]string, len(args))
 			copy(pluginNames, args)
 
+			// Downloads manifest file from given URL
+			if *manifestURL != "" {
```

Comment:
I wonder if you could find a way to not make the `install` function even longer. It already is quite long and is very hard to test. Adding this here only makes it worse. So if you find a way to move this code into some helper where it can be tested independently, that would be great!

Reply:
Unfortunately no. We cannot shift this piece of code to any helper function because we use `defer` to cleanup the temporary files after installation here (https://github.com/rewanth1997/krew/blob/manifest-url/cmd/krew/cmd/install.go#L74)

If we move the piece of code to another function, `defer` will work locally to the particular helper function only and will loose its primary purpose.

Reply:
I tried to make the code as minimal as possible in this section. Its only 5 lines of code.
```
downloadFile()
cleanup()
check_err
assign temporary file name to the manifest variable
```

Reply:
Sorry I'm jumping on this but why do we need to write the temporary file onto the disk? 

Maybe if we changed the installation parameter with an `io.ReadCloser` we can solve this more cleanly?

Reply:
My initial idea was to the downloaded data to a temporary file on the disk and then to pass the temporary file path as a value to `--manifest` variable.

Thanks for pointing the `ReadCloser` interface. This will definitely make the code look cleaner. I can directly save the downloaded file in a variable/buffer and pass it to respective variables rather than storing on disk and removing it later. I will make the changes asap.